### PR TITLE
V3

### PR DESCRIPTION
**Pull request by ArielMAJ**
From my tests (both locally and with an edited version of your GH workflows), this should work properly in most situations. I didn't think up too many edge cases, so it still may need some testing. 

Lmk if there's something this action originally did and I forgot to implement, or anything lacking at all.

I left a lot of comments in assert.py explaining what each thing is doing. But lmk if there's anything that's not clear anyway.

### CODE REVIEW

File: assert.py [hunk 1]
```diff
@@ -1,16 +1,97 @@
...
-print(f"CONTAINS: {input_contains}")
-print(f"EXPECTED RESULT: {input_expected_result}")
+
+ENV_FILE = os.getenv('GITHUB_ENV')
+
+
+def main():
+	# This file should only run in the following conditions:
+	# It should receive exactly 6 arguments (including the file path).
+	if len(sys.argv) != 6:
```
Comment (GuillaumeFalourd) [inline assert.py:11]: All the action inputs aren't required (mandatory) as you can check on the [action.yml file](https://github.com/GuillaumeFalourd/assert-command-line-output/blob/v3/action.yml) therefore I would remove this condition, e eventually just check the mandatory ones. What do you think?

File: assert.py [hunk 2]
```diff
@@ -1,16 +1,97 @@
...
+		sys.exit(
+			f"Invalid usage: not enough arguments. \"Empty\" arguments should receive \"_\".\n\
+			\rExpected:\n{sys.argv[0]} \"<CommandLine>\" \"_<AssertFilePath>\" \"_<SpecificLineNumber>\" \"_<Contains>\" \"<ExpectedResult>\"\n"
+		)
+	# The third, fourth and fifth argument should always start with "_", which will
+	# be ignored and removed from the string later.
+	if not ((sys.argv[2][0] == '_') and (sys.argv[3][0] == '_') and (sys.argv[4][0] == '_')):
+		sys.exit(
+			f"Invalid usage: arguments in the middle should have a \"_\" before the actual input.\n\
+			\rExpected:\n{sys.argv[0]} \"<CommandLine>\" \"_<AssertFilePath>\" \"_<SpecificLineNumber>\" \"_<Contains>\" \"<ExpectedResult>\""
```
Comment (GuillaumeFalourd) [inline assert.py:21]: Could you explain to me why those inputs should start with `_`? I'm asking because you're replacing the `_` afterwards, therefore it doesn't seem necessary to me.

**Review by GuillaumeFalourd — CHANGES_REQUESTED** [review]
Thank you for the contribution @ArielMAJ! 👍🏼 
I wrote some comments, I would be grateful if you could explained to me some of the points I highlighted.

File: assert.py (hunk 1 above)
Comment (ArielMAJ) [reply assert.py:11]: They still aren't required (by the workflow. It's just the script call that needs the workflow to pass in every argument). This is just to ensure all 6 sys.argv elements exist in the script. This required me to change the script call in the action.yml file though.

I ended up having to set this condition because the non-mandatory empty inputs turn into empty strings when we pass them in to "python assert.py <... ... ... ... ...>" and these empty strings are ignored by sys.argv. That ends up turning into a list index out of range error when accessing sys.argv the way the script currently is. It also becomes a bit troublesome (though not impossible) to figure out what options user chose, as what's at each index will change depending on which option the user left empty.

I figured it would be a lot easier to code up and straightforward for readers to understand if we enforce the existence of those outputs by adding any character (such as "_") before the users input in the script call (and later removing that character once the script is run). That way, we still get the same user input, what was left empty in the workflow call will be empty strings in the script, but this way we will be sure that every argument exist (at least as empty strings) and are always in the same position.

Someone using this action would still use it normally. It's the action itself that adds the "_" to the non-mandatory inputs (ensuring all 5 inputs exist) and the script removes them so we still get the actual user input.

File: assert.py (hunk 2 above)
Comment (ArielMAJ) [reply assert.py:21]: Yeah, it's intentionally redundant. As explained above, as some inputs aren't mandatory, they become empty strings in the script call. Those empty strings seem to be ignored by python (or sys.argv I'm not sure).

<hr>
Ex.:

```
python ./assert.py "echo Hi" "" "" "Hi" "PASSED"
sys.argv should be: ["./assert.py", "echo Hi", "", "", "Hi", "PASSED"]
But it actually is: ["./assert.py", "echo Hi", "Hi", "PASSED"]
```

```
python ./assert.py "echo Hi" "./hi.txt" "1" "" "PASSED"
sys.argv should be: ["./assert.py", "echo Hi", "./hi.txt", "1", "", "PASSED"]
But it actually is: ["./assert.py", "echo Hi", "./hi.txt", "1", "PASSED"]
```
<hr>

Note that in that in the first situation sys.argv[2] is the input for "contains" and in the second situation sys.argv[2] is the input for assert_file_path.

For this somewhat small amount of inputs it's not too hard to figure out whether sys.argv[2] is a file path or a normal text. "PASSED" is always at the end of the list which is also easy to find. But what if the user's program outputs a path and the user wants to know if the output "contains" that path? This script would then try to read from a path that doesn't exist. (I did think up some other little bugs when I was writing the logic for this but I already forgot most of them).

That's why I thought it would be easier and more straightforward to add this redundant character. The user doesn't input it, it's the workflow itself that adds it and the script removes it. It just ensures every argument exist.

```
The workflow calls:  python ./assert.py "$command_line" "_$assert_file_path" "_$specific_line" "_$contains" "PASSED"
Which may turn into: python ./assert.py "echo Hi" "_" "_" "_Hi" "PASSED"

sys.argv becomes: ["./assert.py", "echo Hi", "_", "_", "_Hi", "PASSED"]
And then I process it to become:  ["./assert.py", "echo Hi", "", "", "Hi", "PASSED"]
```

Comment (ArielMAJ) [reply assert.py:21]: I was testing this behavior in C and it's the same. If you try to pass command line inputs as _empty strings_, they are ignored (basically the same behavior described above).

**Review by GuillaumeFalourd — APPROVED** [review]
Ok, thanks for the explanations! Let's try it then 😄
